File: ReleaseNotes.Generator/Domain/ProjectTrackingTool.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using ReleaseNotes.Generator.Enums;

namespace ReleaseNotes.Generator.Domain
{
    public class ProjectTrackingTool : EntityBase<int>
    {
        [Required]
        public string Name { get; set; }

        [Required]
        public string Url { get; set; }        

        [Required]
        public string ProjectName { get; set; }

        [Required]
        public ProjectTrackingToolType Type { get; set; }

        public List<Repository> Repositories { get; set; }
    }
}

File: ReleaseNotesGenerator/ClientApp/app/common/classes/ProjectTrackingTool.ts

    import { ProjectTrackingToolType } from './ProjectTrackingToolType';
import { Repository } from './Repository';
import { EntityBase } from './EntityBase';
    export class ProjectTrackingTool extends EntityBase<number> {
        
        name: string;
        url: string;
        projectName: string;
        type: ProjectTrackingToolType;
        repositories: Repository[];
        }

    